src/stream-tcp-inline.c

```diff
@@ -41,9 +41,14 @@ extern int stream_inline;
  *  \retval 0 no
  *  \retval 1 yes
  */
-int StreamTcpInlineMode(void)
+int StreamTcpInlineMode(const Packet *p)
 {
-    return stream_inline;
+    if (stream_inline == PKT_MODE_IDS)
+        return 0;
+    else if (stream_inline == PKT_MODE_IPS)
+        return 1;
+    else /* implied AUTO */
+        return (stream_inline && PacketModeIsIPS(p));
```

Comment:
"else" implies stream_inline == PKT_MODE_AUTO (which is 2), so whats the point of having "(stream_inline && PacketModeIsIPS(p));" then? stream_inline's value can only be one thing. 2. So the condition is always true and thus not needed.